bitcoin/script/miniscript.h

```diff
@@ -568,6 +606,45 @@ struct Node {
         return {{}, {}};
     }
 
+    internal::TimeLockInfo CalcTimeLockInfo() const {
+        switch (nodetype) {
+            case NodeType::PK_K:
+            case NodeType::PK_H: return internal::TimeLockInfo();
+            case NodeType::OLDER: return internal::TimeLockInfo(k < CTxIn::SEQUENCE_LOCKTIME_TYPE_FLAG, k >= CTxIn::SEQUENCE_LOCKTIME_TYPE_FLAG, false, false, false);
+            case NodeType::AFTER: return internal::TimeLockInfo(false, false, k < LOCKTIME_THRESHOLD, k >= LOCKTIME_THRESHOLD, false);
+            case NodeType::SHA256:
+            case NodeType::RIPEMD160:
+            case NodeType::HASH256:
+            case NodeType::HASH160: return internal::TimeLockInfo();
+            case NodeType::ANDOR: return internal::TimeLockInfo(1, Vector(internal::TimeLockInfo(2, Vector(subs[0]->tl_info, subs[1]->tl_info)), subs[2]->tl_info));
+            case NodeType::AND_V:
+            case NodeType::AND_B: return internal::TimeLockInfo(2, Vector(subs[0]->tl_info, subs[1]->tl_info));
+            case NodeType::OR_B:
+            case NodeType::OR_C:
+            case NodeType::OR_D:
+            case NodeType::OR_I: return internal::TimeLockInfo(1, Vector(subs[0]->tl_info, subs[1]->tl_info));
+            case NodeType::MULTI: return internal::TimeLockInfo();
+            case NodeType::WRAP_A:
+            case NodeType::WRAP_S:
+            case NodeType::WRAP_C:
+            case NodeType::WRAP_D:
+            case NodeType::WRAP_V:
+            case NodeType::WRAP_J:
+            case NodeType::WRAP_N: return subs[0]->tl_info;
+            case NodeType::JUST_1:
+            case NodeType::JUST_0: return internal::TimeLockInfo();
+            case NodeType::THRESH: {
+                std::vector<internal::TimeLockInfo> tl_info_vec;
+                for (const auto& sub : subs) {
+                    tl_info_vec.push_back(sub->tl_info);
+                }
+                return internal::TimeLockInfo(k, tl_info_vec);
```

Comment:
This not seem ergonomic. The way I would do this in rust is to use a function that takes in an iterator as an argument. Is this the idiomatic way to consume a vector? 
In the current implementation, we iterate over the loop twice which is not ideal, but on the other hand, I also like that all the timelock boolean logic is inside one function for all fragments.

Reply:
```c++
std::vector<internal::TimeLockInfo> tl_info_vec;
std::transform(subs.begin(), subs.end(), std::back_inserter(tl_info_vec), [](const NodeRef<Key>& arg) { return arg->tl_info; });
return tl_info_vec;
```

perhaps?